### PR TITLE
[KYUUBI #6007] AuthZ should check hoodie procedures path resource privileges

### PR DESCRIPTION
**Pull request by Yikf**
# :mag: Description
## Issue References 🔗

This pull request aims to make authz check hoodie procedures path resource privileges.

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Behavior Without This Pull Request :coffin:
When the Hoodie procedure operation is on the path, the check can pass regardless of whether the path resource has permissions.

#### Behavior With This Pull Request :tada:
Check the path permissions correctly.

#### Related Unit Tests
New tests added.

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by Yikf** [comment]
cc @AngersZhuuuu Could you please take a look if you find a time 🚶 ?

File: extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json [hunk 1]
```diff
@@ -2033,7 +2033,17 @@
   } ],
   "opType" : "QUERY",
   "queryDescs" : [ ],
-  "uriDescs" : [ ]
+  "uriDescs" : [ {
+    "fieldName" : "clone",
+    "fieldExtractor" : "HudiCallProcedureInputUriExtractor",
+    "isInput" : true,
+    "comment" : "Hudi"
```
Comment (AngersZhuuuu) [inline extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:2040]: I think we don't need to add this here, better in the whole hudi node.

File: extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json [hunk 2]
```diff
@@ -2033,7 +2033,17 @@
...
+  }, {
+    "fieldName" : "clone",
+    "fieldExtractor" : "HudiCallProcedureOutputUriExtractor",
+    "isInput" : false,
+    "comment" : "Hudi"
```
Comment (AngersZhuuuu) [inline extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:2045]: ditto

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/HudiCommands.scala [hunk 3]
```diff
@@ -288,7 +288,14 @@ object HudiCommands extends CommandSpecs[TableCommandSpec] {
...
           actionTypeDesc = Some(ActionTypeDesc(actionType = Some(UPDATE))),
-          setCurrentDatabaseIfMissing = true)))
+          setCurrentDatabaseIfMissing = true)),
+      uriDescs = Seq(
+        UriDesc(
+          "clone",
+          classOf[HudiCallProcedureInputUriExtractor],
+          isInput = true,
+          comment = "Hudi"),
+        UriDesc("clone", classOf[HudiCallProcedureOutputUriExtractor], comment = "Hudi")))
```
Comment (AngersZhuuuu) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/gen/HudiCommands.scala:298]: ditto, can remove hudi

**Review by AngersZhuuuu — COMMENTED** [review]
Good catch!

File: extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json (hunk 1 above)
Comment (Yikf) [reply extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:2040]: thanks @AngersZhuuuu for the suggestion.

Maybe I misunderstood your meaning, do you mean the 'comment' attribute that can be removed?

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:2040]: > thanks @AngersZhuuuu for the suggestion.
> 
> Maybe I misunderstood your meaning, do you mean the 'comment' attribute that can be removed?

Yea, if you want to add comment to define query type, we can add it in top level 
``` {
  "classname" : "org.apache.spark.sql.hudi.command.CallProcedureHoodieCommand",
  "comment": "Hudi"
````

**Comment by Yikf** [comment]
> Pls create a issue for this and change your pr title

Sure

File: extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json (hunk 1 above)
Comment (Yikf) [reply extensions/spark/kyuubi-spark-authz/src/main/resources/table_command_spec.json:2040]: As far as I know, the comment field is only used for [error output](https://github.com/apache/kyuubi/blob/ac388d191ddb6a910a5c74575d3ba6aa5a31e062/extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/Descriptor.scala#L64)(maybe I missed something 👽), So I don't understand what the purpose of deleting this field is?

BTW, the top level CommandSpec has no comment field.

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

**Comment by Yikf** [comment]
thanks, merged to master

**Pull request closed without merging**